src/WebViewShared.tsx

```diff
@@ -16,29 +16,46 @@ import {
 import styles from './WebView.styles';
 
 const defaultOriginWhitelist = ['https://*'] as const;
+const defaultDeeplinkWhitelist = ['https://'] as const;
+const defaultDeeplinkBlocklist = [/^http:/i, /^file:/i, /^javascript:/i] as const;
 
 const extractOrigin = (url: string): string => {
   const result = /^[A-Za-z][A-Za-z0-9+\-.]+:(\/\/)?[^/]*/.exec(url);
   return result === null ? '' : result[0];
 };
 
-const originWhitelistToRegex = (originWhitelist: string): string =>
-  `^${escapeStringRegexp(originWhitelist).replace(/\\\*/g, '.*')}$`;
+const stringWhitelistToRegex = (originWhitelist: string): RegExp =>
+  new RegExp(`^${escapeStringRegexp(originWhitelist).replace(/\\\*/g, '.*')}$`);
+
+const matchWithRegexList = (
+  compiledRegexList: readonly RegExp[],
+  value: string,
+) => {
+  return compiledRegexList.some(x => x.test(value));
+};
+
+const matchWithPrefixStringList = (
+  prefixes: readonly string[],
+  value: string,
+) => {
+  if (typeof value !== 'string') throw new Error(`value was not a string`)
+  return prefixes.some(x => String(x).length && String.prototype.startsWith.call(value, x));
```

Comment:
Would be good to add tests for [these](https://github.com/ExodusMovement/react-native-webview/pull/21#discussion_r1619322987) cases.